docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
+compute and remote execution. You can also save a lot of money by reusing compute across multiple nodes or reusing
+services across multiple pipelines with Runhouse, which is generally not possible with workflow orchestrators.
+
+<diagram showing an Airflow pipeline calling multiple steps on one cluster, and another pipeline calling a service
+created by the first>
+
+*Serverless frameworks* like Modal or AWS Lambda allow for the allocation of services on the fly, but within a well
+defined sandbox, and not strictly from within regular Python - they require specific pre-packaging or CLI launch
+commands outside Python. Runhouse runs fully in a Python interpreter so it can extend the compute power of practically
+any existing Python application, and allocates services inside your own compute, wherever that may be. We may even
+support serverless systems as compute backends in the future.
+
+<diagram showing Modal creating new services within their own compute, and Runhouse within EC2, K8s, GCP, etc.>
+
+*Infrastructure in code* tools like SkyPilot or Pulumi allocate compute on the fly, but can't utilize it instantly
+to offload execution within the application (though you could call a predefined script entrypoint or API
+endpoint). Runhouse uses SkyPilot to allocate compute, but is vertically integrated to be able
+to perform allocation, (re)deployment, and management of a new service all in Python so the new compute can be used
+instantly within the existing application. It also doesn't need to perform allocation to create new services -
+it can use existing compute or static VMs.
+
+<diagram showing SkyPilot running a CLI command on the remote compute and getting back logs, and Runhouse using
+SkyPilot to launch new compute, send a function to it, and call it>
+
+*GPU/Accelerator dispatch* systems like PyTorch, Jax, or Mojo give the ability to offload computation to a local GPU or
+TPU. Runhouse does not have this capability, but can offload a function or class to a remote instance with an
+accelerator, which can then itself use PyTorch or Jax (and maybe one day Mojo) to use the accelerator.
+
+<diagram showing PyTorch dispatching to the GPU within the node, Runhouse dispatching to a remote node which then uses
+PyTorch to dispatch to the GPU>
+
+Saving, Loading, and Sharing
+~~~~~~~~~~~~~~~~~~
 
-* **Organization structure**: Both resources and users can be organized into organizations to apply and manage shared
-  access and permissions. Resources can be created for an organization by specifying its name in full
-  when creating or loading down a Runhouse resource.
+Runhouse resources (clusters, functions, modules, environments) can be saved, shared, and reused based on a compact
+JSON metadata signature. This allows for easy sharing of clusters and services across users and environments, which
+can often lead to massive cost savings. The `Den <https://www.run.house/dashboard>`_ hosted metadata store can be
```

Comment:
I think we could expand on what Den is here. A full sentence explaining before saying how to access maybe?